sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/SecurityTestApp.java

```diff
@@ -7,52 +7,46 @@
  */
 package org.sdase.commons.spring.boot.web.security.test;
 
-import java.io.IOException;
-import java.time.LocalDateTime;
-import java.time.ZoneOffset;
-import java.time.ZonedDateTime;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+import org.sdase.commons.spring.boot.web.error.ApiException;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 @SpringBootApplication
 @RestController
 public class SecurityTestApp {
 
-  @GetMapping("/fixedTime")
-  public Object getFixedTime() {
-    return new Object() {
-      private final ZonedDateTime time =
-          // the configured ObjectMapper should truncate the nanos to seconds
-          ZonedDateTime.of(LocalDateTime.of(2018, 11, 21, 13, 16, 47, 965_000_300), ZoneOffset.UTC);
-
-      @SuppressWarnings("unused")
-      public ZonedDateTime getTime() {
-        return time;
-      }
-    };
-  }
-
   @RequestMapping(value = "/traced", method = RequestMethod.TRACE)
   public String serveTrace() {
     return "This should not be allowed";
   }
 
-  @GetMapping(value = "/error")
-  public Object throwError() throws IOException {
-    throw new IOException("This should not be allowed");
+  @GetMapping(value = "/forcedError")
+  public Object throwError() {
+    throw new RuntimeException("This should not be allowed");
+  }
+
+  @GetMapping(value = "/apiError")
+  public void throwApiError() {
+    throw ApiException.builder()
+        .httpCode(HttpStatus.NOT_IMPLEMENTED.value())
+        .title("This method is not implemented yet.")
+        .build();
   }
 
   @GetMapping(value = "/errorEntity")
   public ResponseEntity<?> responseError() {
-    var body =
-        ResponseEntity.internalServerError()
-            .body(new TestResource().setValue("This should not be leaked."));
-    return body;
+    return ResponseEntity.internalServerError()
+        .body(new TestResource().setValue("This should not be leaked."));
   }
 
   @GetMapping(value = "/response")
```

Comment:
@saadmar : I'd propose to change the returned status code to 200. Does that break any test as far as you remember?

Reply:
I guess only [this one](https://github.com/SDA-SE/sda-spring-boot-commons/blob/d36a57470016055ccfe9ea0c0ee55f71e1e62816/sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java#L101) that you just pointed above

Reply:
fixed in #162